src/index.js

```diff
@@ -31,7 +42,12 @@ export default async function (moduleOptions) {
         throw new Error(formatMessage(`All locales must be objects and have the "file" property set when using "langDir".\nFound none in:\n${JSON.stringify(locale, null, 2)}.`))
       }
     }
-    options.langDir = this.nuxt.resolver.resolveAlias(options.langDir)
+    const resolvedLangDir = this.nuxt.resolver.resolveAlias(options.langDir)
+    for (const locale of options.locales) {
+      if (typeof locale !== 'string' && locale.file) {
+        localeFileMap[String(locale.file)] = relativeToBuild(resolvedLangDir, locale.file)
+      }
+    }
```

Comment:
Not that important but could do with less duplication and better variable scoping with something like:


```js
  /** @type {Record<string, string>} */
  const localeFileMap = {}

  if (options.langDir) {
    /**
     * Helper for `langDir` resources resolving
     * @param {any[]} args
     */
    const relativeToBuild = (...args) => relativeTo(this.options.buildDir, ...args)
    const resolvedLangDir = this.nuxt.resolver.resolveAlias(options.langDir)

    if (!options.locales.length || typeof options.locales[0] === 'string') {
      throw new Error(formatMessage('When using the "langDir" option the "locales" must be a list of objects.'))
    }
    for (const locale of options.locales) {
      if (typeof (locale) === 'string' || !locale.file) {
        throw new Error(formatMessage(`All locales must be objects and have the "file" property set when using "langDir".\nFound none in:\n${JSON.stringify(locale, null, 2)}.`))
      }
      localeFileMap[String(locale.file)] = relativeToBuild(resolvedLangDir, locale.file)
    }
  }
```